uber/templates/preregistration/stripeForm.html

```diff
@@ -1,16 +1,161 @@
-        data-allow-remember-me="false"
-        data-amount="{{ charge.amount }}"
-        data-name="{{ c.EVENT_NAME_AND_YEAR }} {{ regtext }}"
-        data-description="{{ charge.description }}"
-        data-image="../static/theme/stripe-logo.png">
-    </script>
-</form>
+
+<style>
+.StripeElement {
+  box-sizing: border-box;
+
+  height: 40px;
+
+  padding: 10px 12px;
+
+  border: 1px solid transparent;
+  border-radius: 4px;
+  background-color: white;
+
+  box-shadow: 0 1px 3px 0 #e6ebf1;
+  -webkit-transition: box-shadow 150ms ease;
+  transition: box-shadow 150ms ease;
+}
+
+.StripeElement--focus {
+  box-shadow: 0 1px 3px 0 #cfd7df;
+}
+
+.StripeElement--invalid {
+  border-color: #fa755a;
+}
+
+.StripeElement--webkit-autofill {
+  background-color: #fefde5 !important;
+}
+</style>
+<button class="btn btn-info" onClick="callStripeAction();">
+<span class="display: block; min-height: 30px;">Pay with Card</span>
+</button>
+<div id="stripeModal" class="modal" style="color:black;" tabindex="-1" role="dialog" data-backdrop="static">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+        <div class="modal-header">
+        <h4 class="modal-title text-center"></h4>
+        </div>
+        <div class="modal-body text-left jumbotron">
+        <div class="charge_desc" style="margin-bottom:10px;"></div>
+        <form action="/charge" method="post" id="payment-form" class="hidden">
+        <div class="form-row">
+            <label for="card-element">
+            Credit or debit card
+            </label>
+            <div id="card-element">
+            <!-- A Stripe Element will be inserted here. -->
+            </div>
+
+            <!-- Used to display form errors. -->
+            <div id="card-errors" role="alert"></div>
+        </div>
+        <div style="margin-top:10px;"><button class="btn btn-success">Submit Payment</button>
+        <button class="btn btn-default" data-dismiss="modal">Cancel</button></div>
+        </form>
+        </div>
+    </div>
+  </div>
+</div>
+<script type="text/javascript">
+stripe_action_params = {id: '{{ id }}'};
+
+var callStripeAction = function () {
+    var $btn = $(this);
+    $btn.prop('disabled', true);
+    {% for key, val in params.items() %}
+        stripe_action_params['{{ key }}'] = '{{ val }}';
+    {% endfor %}
+    stripe_action_params.csrf_token = csrf_token;
+    $.post('{{ action }}', stripe_action_params, 
+    function(result) {
+        $btn.prop('disabled', false);
+        if (result.error) {
+            toastr.error(result.error);
+        } else if (result.stripe_intent) {
+            var dollars = result.stripe_intent.amount / 100;
+            dollars = dollars.toLocaleString("en-US", {style:"currency", currency:"USD"});
+            var cancel_url = result.cancel_url || '../preregistration/cancel_payment'
+            $('#stripeModal').find('.modal-title').text('Paying ' + dollars + ' to {{ c.ORGANIZATION_NAME }}');
```

Comment:
Surprisingly, Stripe Elements doesn't natively include a display the amount of money someone is being charged or the description of the payment intent.